File: frontend/lib/core/models/alerta.dart
enum TipoAlerta { RESTAURANTE, PRODUCTO, COMENTARIO }

enum EstadoAlerta { PENDIENTE, ACEPTADA, DENEGADA }

TipoAlerta tipoAlertaDesde(String v) => TipoAlerta.values.firstWhere((e) => e.name == v);
EstadoAlerta estadoAlertaDesde(String v) => EstadoAlerta.values.firstWhere((e) => e.name == v);

class Alerta {
  final int id;
  final String? usuarioNombre;
  final TipoAlerta tipo;
  final int? restauranteId;
  final String? restauranteNombre;
  final int? productoId;
  final String? productoNombre;
  final String descripcion;
  final EstadoAlerta estado;
  final DateTime fecha;

  Alerta({
    required this.id,
    this.usuarioNombre,
    required this.tipo,
    this.restauranteId,
    this.restauranteNombre,
    this.productoId,
    this.productoNombre,
    required this.descripcion,
    required this.estado,
    required this.fecha,
  });

  factory Alerta.fromJson(Map<String, dynamic> json) => Alerta(
        id: json['id'],
        usuarioNombre: json['usuarioNombre'],
        tipo: tipoAlertaDesde(json['tipo']),
        restauranteId: json['restauranteId'],
        restauranteNombre: json['restauranteNombre'],
        productoId: json['productoId'],
        productoNombre: json['productoNombre'],
        descripcion: json['descripcion'] ?? '',
        estado: estadoAlertaDesde(json['estado']),
        fecha: DateTime.tryParse(json['fecha'] ?? '') ?? DateTime.now(),
      );
}

File: backend/src/main/java/com/safebite/backend/model/Alerta.java
package com.safebite.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "alertas")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Alerta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "usuario_id")
    private Usuario usuario;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TipoAlerta tipo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "restaurante_id")
    private Restaurante restaurante;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "producto_id")
    private Producto producto;

    @Column(nullable = false, length = 1000)
    private String descripcion;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private EstadoAlerta estado = EstadoAlerta.PENDIENTE;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "revisado_por")
    private Usuario revisadoPor;

    @Builder.Default
    private LocalDateTime fecha = LocalDateTime.now();

    private LocalDateTime fechaRevision;
}
